File: languages/go/test-one.go
// The test-one-go tool parses the provided HTML file and print the microformats found.
package main

import (
	"encoding/json"
	"fmt"
	"net/url"
	"os"

	"willnorris.com/go/microformats"
)

func main() {
	f, err := os.Open(os.Args[1])
	if err != nil {
		fmt.Println(err)
		os.Exit(1)
	}
	defer f.Close()

	baseURL, _ := url.Parse("http://example.com/")
	data := microformats.Parse(f, baseURL)

	json, _ := json.MarshalIndent(data, "", "  ")
	fmt.Println(string(json))
}

File: languages/typescript/test-one.js
const { mf2 } = require('microformats-parser')

var infile = process.argv[2];

if (infile === undefined) {
  console.log("Usage: " + process.argv[1] + " <inputfile>");
  process.exit(1);
}

var fs = require("fs");

fs.readFile(infile, "utf8", function (err, data) {
    if (err) throw err;

    const parsed = mf2(data, {
        baseUrl: "http://example.com/",
    });
    console.log("%j", parsed);
});

